libraries-server-2/src/main/java/com/baeldung/netty/customhandlersandlisteners/ChatClientMain.java

```diff
@@ -0,0 +1,75 @@
+package com.baeldung.netty.customhandlersandlisteners;
+
+import java.util.Scanner;
+
+import com.baeldung.netty.customhandlersandlisteners.handler.ClientEventHandler;
+import com.baeldung.netty.customhandlersandlisteners.listener.ChannelInfoListener;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.string.StringDecoder;
+import io.netty.handler.codec.string.StringEncoder;
+
+public class ChatClientMain {
+
+    private static final String SYSTEM_USER = System.getProperty("user.name");
+    private static String user;
+
+    public static void main(String[] args) {
+        EventLoopGroup group = new NioEventLoopGroup();
+        try (Scanner scanner = new Scanner(System.in)) {
+            Bootstrap bootstrap = new Bootstrap();
+            bootstrap.group(group)
+                .channel(NioSocketChannel.class)
+                .handler(new ChannelInitializer<SocketChannel>() {
+                    @Override
+                    public void initChannel(SocketChannel channel) throws Exception {
+                        channel.pipeline()
+                            .addFirst(new StringDecoder(), new ClientEventHandler(), new StringEncoder());
+                    }
+                });
+
+            ChannelFuture future = bootstrap.connect(ChatServerMain.HOST, ChatServerMain.PORT)
+                .sync();
+
+            future.addListener(new ChannelInfoListener());
+            Channel channel = future.sync()
+                .channel();
+
+            messageLoop(scanner, channel);
+
+            channel.close();
+        } catch (Throwable e) {
+            e.printStackTrace();
```

Comment:
We should handle the exception rethrowing it or interrupting the thread.
```suggestion
            e.printStackTrace();
            Thread.currentThread().interrupt();
```